src/client/opamSwitchCommand.ml

```diff
@@ -495,6 +495,18 @@ let import st filename =
   in
   import_t importfile st
 
+let set_compiler st names =
+  let packages =
+    try List.map (OpamSwitchState.find_installed_package_by_name st) names
+    with Not_found ->
+      OpamConsole.error_and_exit "These packages are not installed: %s"
+        (OpamStd.List.concat_map ", " OpamPackage.Name.to_string
+           (List.filter (not @* OpamSwitchState.is_name_installed st) names))
+  in
+  let st = { st with compiler_packages = OpamPackage.Set.of_list packages } in
```

Comment:
I'm working on a related matter - shouldn't this step also compute the transitive closure of the dependencies of the listed packages? Otherwise the cudf universe has packages which are really base packages depending on other base packages. I'm not sure that that creates a problem at the moment, though it certainly creates a problem in my fork where you can explicitly upgrade base packages.

I'm working on a related issue at the moment: upgrading a compiler package (in this case tcl/tk) which itself draws in a new dependent package (upgrading Tcl/Tk from 8.5 to 8.6 adds a requirement for ZLib)